src/Umbraco.Web.BackOffice/Trees/TreeControllerBase.cs

```diff
@@ -0,0 +1,399 @@
+﻿using System;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Umbraco.Core;
+using Umbraco.Core.Events;
+using Umbraco.Core.Models;
+using Umbraco.Core.Models.Entities;
+using Umbraco.Core.Persistence;
+using Umbraco.Web.BackOffice.Controllers;
+using Umbraco.Web.Common.Filters;
+using Umbraco.Web.Common.ModelBinders;
+using Umbraco.Web.Models.Trees;
+using Umbraco.Web.Trees;
+using Umbraco.Web.WebApi;
+
+namespace Umbraco.Web.BackOffice.Trees
+{
+    /// <summary>
+    /// A base controller reference for non-attributed trees (un-registered).
+    /// </summary>
+    /// <remarks>
+    /// Developers should generally inherit from TreeController.
+    /// </remarks>
+    [TypeFilter(typeof(AngularJsonOnlyConfigurationAttribute))]
+    public abstract class TreeControllerBase : UmbracoAuthorizedApiController, ITree
+    {
+        // TODO: Need to set this, but from where?
+        //       Presumably not injecting as this will be a base controller for package/solution developers.
+        private readonly UmbracoApiControllerTypeCollection _apiControllers;
```

Comment:
This is the point I'm stuck on - where to do get this from?  It's needed to be passed into the `GetUmbracoApiService` method in `UrlHelperExtensions`.